rest-api/src/main/java/gov/cms/qpp/conversion/api/services/CpcFileServiceImpl.java

```diff
@@ -62,13 +65,17 @@ public InputStreamResource getFileById(String fileId) throws IOException {
 	 */
 	public String processFileById(String fileId) {
 		Metadata metadata = dbService.getMetadataById(fileId);
-		if (isAnUnprocessedCpcFile(metadata)) {
+		if (metadata == null) {
+			throw new NoFileInDatabaseException(FILE_NOT_FOUND);
+		} else if (!metadata.getCpc()) {
+			throw new InvalidFileTypeException(INVALID_FILE);
+		} else if (metadata.getCpcProcessed()) {
+			return PROCESSED;
```

Comment:
I think a `4xx` HTTP status should be sent back for any scenario that doesn't result in a file being marked as processed.  If we don't, then we are sending conflicting information back: that nothing happened (the text body we send back) but that the request was successful (the `200` HTTP status we send).

Thoughts @sfradkin?

Reply:
Or does anyone else have a thought on this?  @omnivion @clydet?

Reply:
Updating this value should probably be a PUT request.  And if it's a PUT request, then the request should be idempotent.  Meaning that subsequent requests that are the same should have the same result.  With that thinking, a CPC+ file that is already marked as processed will remain marked as processed.  It's not an error, it just doesn't do anything because it's already been done.

Reply:
Oh yeah, very good point about being idempotent.  I retract my issue.  But instead of returning `PROCESSED`, it should return `FILE_FOUND` again.  Agreed on the this being a PUT instead of a POST too.

Reply:
Okay. I changed this to be a PUT request and removed the extra messaging. I will also have to get KONG to allow PUT requests on our endpoints as well

Reply:
I went ahead and commented on https://jira.cms.gov/browse/QPPFED-532 to allow PUT method.